src/components/Plans/PlanCalculator/index.js

```diff
@@ -98,11 +100,13 @@ export const PlanCalculator = InjectAppServices(
       setSelectedPlanIndex(_selectedPlanIndex);
       dispatchPlansByType({
         type: PLANS_BY_TYPE_ACTIONS.SEARCH_DISCOUNTS_BY_INDEX_PLAN,
-        payload: _selectedPlanIndex,
+        payload: { _selectedPlanIndex, selectedDiscountIndex },
```

Comment:
Why do you need to send `selectedDiscountIndex` when the event type is `SEARCH_DISCOUNTS_BY_INDEX_PLAN`? In my opinion, the natural place to send it is in the event `CHANGE_SELECTED_DISCOUNT`